character.cpp

```diff
@@ -0,0 +1,39 @@
+#include "character.h"
+
+Character::Character(std::string name, int health, unsigned int damage) :
+                    name(name),
+                    health(health),
+                    damage(damage) {}
+
+std::string Character::GetName() const{
+  return name;
+}
+
+int Character::GetHealth() const{
+  return health;
+}
+
+unsigned int Character::GetDamage() const{
+  return damage;
+}
+
+bool Character::IsAlive(){
+  return health > 0;
+}
+
+void Character::TakeDamage(const int damage){
```

Comment:
tul altalanos, nem akarjuk, hogy barmikor barmennyivel csokkentheto legyen. Azt akarjuk csak, hogy meg lehessen utni masik karakterrel.

Reply:
ezt megkérdezhetem hogy miért túl általános? legyen helyette egy setHealth függvény?

Reply:
azzal pont ugyanaz lenne a baj, mint amit itt leirtam. `u1.setHealth(1000000000000000)`-et sem akarunk engedni. A masik iranynal (attack) jobb a szignatura ilyen szempontbol, olyan kellene itt is. Kerdes persze, hogy szukseg van-e mindkettore.

Reply:
még mindig nem értem :D Akkor hogyan lehetne úgy sebződni, hogy nincs rá olyan függvény ami az életet csökkenti?

Reply:
Ejj, nem lehet ravezetni, csak elmondani.... Negy random int-et varjon, amivel sebzodik, hanem egy masik Unitot, aki megsebzi, es O kerje le BELUL annak a tamadaserteket. A jelenlegi szignatura megengedi, hogy barhol barki egy unit eletet kenye kedve szerint modositsa tetszoleges ertekkel. IGy csak egy masik unit tudja megutni. Amit irtam meg, az az, hogy az attack meg ez a fuggveny ket iranya ugyanannak, nem feltetlen szukseges mindketo.

Reply:
A lenyeg itt meg mindig nem jott at, akkor most mar expliciten leirom. `bool/void takeDamageFrom(const Character&)`. NEm a fuggvenytorzzsel van baj, hanem azzal, hogy a Te valtozatodban barmennyivel csokkenthetem a hp-t. megadok 12329840-at az argumentumnak es csokkenti, hiaba nincs is olyan leny, akinek ennyi lenne a sebzese. Ezt irtam le fent parszor.